packages/contracts/test/EmailSigner.t.sol

```diff
@@ -0,0 +1,155 @@
+import "./helpers/SignerStructHelper.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {IEmailAuth} from "../src/interfaces/IEmailAuth.sol";
+import {IEmailAuthErrors} from "../src/interfaces/IEmailAuth.sol";
+import {ERC1271} from "../src/libraries/ERC1271.sol";
+
+contract EmailSignerTest is SignerStructHelper, IEmailAuthErrors {
+    string message = "message to sign";
+    bytes32 msgHash = keccak256(abi.encode(message));
+
+    function testDkimRegistryAddr() public view {
+        address dkimAddr = emailSigner.dkimRegistryAddr();
+        assertEq(dkimAddr, address(dkim));
+    }
+
+    function testVerifierAddr() public view {
+        address verifierAddr = emailSigner.verifierAddr();
+        assertEq(verifierAddr, address(verifier));
+    }
+
+    function testVerifyEmail() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+        emailSigner.verifyEmail(emailAuthMsg); // should not revert
+    }
+
+    function testExpectRevertVerifyEmailInvalidDkimPublicKeyHash() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+        emailAuthMsg.proof.domainName = "invalid.com";
+
+        vm.expectRevert(InvalidDKIMPublicKeyHash.selector);
+        emailSigner.verifyEmail(emailAuthMsg);
+    }
+
+    function testExpectRevertVerifyEmailInvalidAccountSalt() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+        emailAuthMsg.proof.accountSalt = bytes32(uint256(1234));
+
+        vm.expectRevert(InvalidAccountSalt.selector);
+        emailSigner.verifyEmail(emailAuthMsg);
+    }
+
+    function testExpectRevertVerifyEmailInvalidCommand() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+        emailAuthMsg.commandParams[0] = abi.encode(2 ether);
+
+        vm.expectRevert(InvalidCommand.selector);
+        emailSigner.verifyEmail(emailAuthMsg);
+    }
+
+    function testExpectRevertVerifyEmailInvalidEmailProof() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+
+        vm.mockCall(
+            address(verifier),
+            abi.encodeWithSelector(
+                Verifier.verifyEmailProof.selector,
+                emailAuthMsg.proof
+            ),
+            abi.encode(false)
+        );
+        vm.expectRevert(InvalidEmailProof.selector);
+        emailSigner.verifyEmail(emailAuthMsg);
+    }
+
+    function testExpectRevertVerifyEmailInvalidMaskedCommandLength() public {
+        EmailAuthMsg memory emailAuthMsg = buildEmailAuthMsg(msgHash);
+
+        // Set masked command length to 606, which should be 605 or less defined in the verifier.
+        emailAuthMsg.proof.maskedCommand = string(new bytes(606));
+
+        vm.expectRevert(InvalidMaskedCommandLength.selector);
+        emailSigner.verifyEmail(emailAuthMsg);
+    }
+
+    function testExpectRevertVerifyEmailInvalidSizeOfTheSkippedCommandPrefix()
```

Comment:
For complete coverage, we should add a test case for if the command prefix is greater than the command bytes, as well as equal